src/shader/shaderUtils.js

```diff
@@ -290,4 +296,28 @@ export const shaderUtils = /* glsl */`
 
 	}
 
+	float square( float t ) {
+
+		return t * t;
+
+	}
+
+	vec2 square( vec2 t) {
+
+		return t * t;
+
+	}
+
+	vec3 square( vec3 t ) {
+
+		return t * t;
+
+	}
+
+	vec4 square( vec4 t ) {
+
+		return t * t;
+
+	}
```

Comment:
Out of curiosity - is this faster or better than using `pow( t, 2.0 )`?

Reply:
I believe it will be faster unless the compiler could detect the constant on the RHS and specialises this; however, `pow( x, y )` is undefined for `x < 0` which would make this specialisation invalid.